Modified layout

TheKingWizard opened the pull request:
- Removed Quasar propaganda
- Moved drawer to right side
- Added links to the corresponding pages
- Added button to log in / sign up
- Set ice-doge to be the tab icon

TheKingWizard committed:
Modified layout

- Removed Quasar propaganda
- Moved drawer to right side
- Added links to the corresponding pages
- Added button to log in / sign up
- Set ice-doge to be the tab icon

adnanrh commented:
Nice work. Comments:
* Should the ice doge be by the ICEMEME heading, like it is in the mockups? 
* The home and log in pages in the menu are always highlighted blue, when only the page you're currently on should be highlighted, right?
* I think it's nice to have the username displayed where the sign up/login button is once you're logged in instead of at the top of the menu (like it is in the mockups).

TheKingWizard commented:
- We could put the ice-doge by the ICEMEME headings. It's actually not in the mockups that I looked at, it was added to the ones in the presentation and I didn't notice.
- The highlight is definitely a bug, I'm not sure what's causing it but I think it has something to do with the routing to the index page. 
- I figured that the sign up/login button would be replaced by the username when a user logged in, but that just isn't implemented yet.

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,116 @@
           tag="a"
-          target="_blank"
-          href="http://v1.quasar-framework.org"
+          :to="{ name: 'home' }"
```
The fix to the highlight issue is to add the `exact` parameter to the `q-item` element for home.
https://router.vuejs.org/api/#exact

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,116 @@
+          >
+            ICEMEME
+          </font>
+        </q-toolbar-title>
```
- don't use a `font` element ([it's obsolete](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/font)), instead define a class on the `q-toolbar-title` that sets the style
- wrap the `q-toolbar-title` in a `router-link` element linking to the home page

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,116 @@
+          :to="{ name: 'register' }"
+        >
+          Sign Up / Log In
+        </q-btn>
```
Because sign up is its own page and login is handled by a dialog (or maybe a dropdown), lets use two separate buttons here.

Also, replace `color="secondary"` with `flat` and `stretch` to make toolbar-style buttons

TheKingWizard replied:
What I had in mind is instead of having two buttons the one button would open the login dialog which would have a link to the sign-up page

nxtman123 replied:
That could also work, I'm just used to seeing both buttons separate. It would also make for fewer clicks to sign up.

Your choice. I'd recommend looking at how other sites do it

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,116 @@
           </q-item-section>
           <q-item-section>
-            <q-item-label>Docs</q-item-label>
+            <q-item-label>Home</q-item-label>
```
Maybe instead of "home" this could say "latest" and use the `whatshot` icon? Just a suggestion, what do you think @adnanrh?

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,116 @@
           tag="a"
-          target="_blank"
-          href="https://twitter.com/quasarframework"
+          :to="{ name: 'home' }"
```
instead of linking `:to=...` someplace, this button should trigger a function. You can [use](https://vuejs.org/v2/guide/syntax.html#v-on-Shorthand) `@click="logout"` for example, and add `logout` as a new function under `methods` in the `<script>` area. Stub it out with an `alert()` to show it's getting called.

TheKingWizard committed:
Merge branch 'develop' into feature/layout-change

TheKingWizard committed:
More minor layout changes

- Fixed side drawer highlight issue
- Removed <font> element and added CSS styling instead
- Linked ICEMEME to home page
- Changed Log in / Sign up styling to fix with the rest of the toolbar
- Changed "Home" to "Latest"
- Linked Log out to a stub function

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -127,11 +135,14 @@ export default {
   methods: {
     openURL,
+    logout() {
+      alert('TODO: LOGOUT');
```
when linting you'll need this
```suggestion
      // eslint-disable-next-line no-alert
      alert('TODO: LOGOUT');
```

nxtman123 commented on src/layouts/MyLayout.vue:
```diff
@@ -4,110 +4,118 @@
+            <q-item-label>Settings</q-item-label>
             <q-item-label caption>
-              forum.quasar-framework.org
+              Profile Settings
```
I'd suggest "Account Settings" because it's just change password and email

nxtman123 reviewed (changes requested):
Make sure to run `yarn lint` to check style and potential bugs. Use `yarn lint --fix` to automatically fix the easy stuff.

If you merge develop into your feature branch, the CI integration I set up will run the lint on this PR

nxtman123 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Modify Layout](https://trello.com/c/ggadLLiZ/11-modify-layout)

TheKingWizard committed:
Merge branch 'develop' into feature/layout-change

TheKingWizard committed:
More minor changes

- changed "Profile Settings" to "Account Settings"
- lint fix for logout stub

TheKingWizard committed:
Merge remote-tracking branch 'origin/develop' into feature/layout-change
